pkg/crc/preflight/preflight_checks_darwin.go

```diff
@@ -261,6 +261,9 @@ func getDaemonConfig() (*launchd.AgentConfig, error) {
 }
 
 func checkIfDaemonPlistFileExists() error {
+	if err := olderDaemonVersionRunning(); err != nil {
```

Comment:
The last if clause will make this check return error if the daemon is not running via `launchd` and `crc setup --check-only` will also return error, it should be modified to consider the case when `daemon` started from the `tray`, similar to windows. something like:
```go
if !launchd.AgentRunning(daemonConfig.Label) || !daemonRunning() {
	return fmt.Errorf("launchd agent '%s' is not running", daemonConfig.Label)
}
```

however, we have enabled the `daemon` to auto-start at boot on macOS also, and while testing `daemon` starts before the `tray`, but in the case where the tray starts first and `launchd` silently fails to start the daemon, we might see the above happening

Reply:
I'm not in favour of doing something similar to what is done on Windows. Preflight checks usually check that the system is in the state we expect. On macOS we expect the daemon to be started with launchd, so it is an error if it's not launched this way, even if the daemon was started some other way. If one does not want to use launchd to start the daemon, then they can set `skip-check-launchd` in their config.

Reply:
@anjannath So plan is to have something similar for windows task scheduler so that task should start as soon as user logs in. From tray side then we can completely remove the daemon starting part.

Reply:
> @anjannath So plan is to have something similar for windows task scheduler so that task should start as soon as user logs in. From tray side then we can completely remove the daemon starting part.

yep, that'll solve the issue i mentioned